src/core/thread/mle_router.cpp

```diff
@@ -519,7 +519,15 @@ bool MleRouter::HandleAdvertiseTimer(void)
         return false;
     }
 
-    SendAdvertisement();
+    // Suppress MLE Advertisements when trying to attach to a better partition.
+    //
+    // Without this suppression, a device may send an MLE Advertisement before receiving the MLE Child ID Response.
+    // The candidate parent then removes the attaching device because the Source Address TLV includes an RLOC16 that
+    // indicates a Router role (i.e. a Child ID equal to zero).
+    if (mParentRequestState == kParentIdle)
```

Comment:
Move this check inside `SendAdvertisement()`?

Reply:
Updated. Thanks.